mono/metadata/appdomain.c

```diff
@@ -1324,6 +1302,24 @@ ves_icall_System_AppDomain_GetAssemblies (MonoAppDomainHandle ad, MonoBoolean re
 	return res;
 }
 
+MonoArrayHandle
+ves_icall_System_AppDomain_GetAssemblies (MonoAppDomainHandle ad, MonoBoolean refonly, MonoError *error)
+{
+	error_init (error);
+	MonoDomain *domain = MONO_HANDLE_GETVAL (ad, data);
+	return get_assembly_array_from_domain (domain, refonly, error);
+}
+
+#ifdef ENABLE_NETCORE
+MonoArrayHandle
+ves_icall_System_Runtime_Loader_AssemblyLoadContext_InternalGetLoadedAssemblies (MonoBoolean refonly, MonoError *error)
+{
+	error_init (error);
+	MonoDomain *domain = mono_domain_get ();
```

Comment:
No need to init the error.

Reply:
I think netcore has no refonly assemblies.

Reply:
Do I not need to init the error for both this and the AppDomain call, or just this one? I'm not sure I understand why it's not needed here.

Reply:
You shouldn't need it in both.

We changed how MonoError works at one point but didn't finish updating the code.  used to be that you had to call `error_init (error)` on entry to any function with a `MonoError *error` argument.

These days you no longer need to do that.  Instead the initialization is done once when the error is declared using `ERROR_DECL (error)`. In icalls it's done with macros when we make the handles wrapper https://github.com/mono/mono/blob/090dffb9a09f26dc250d8d09a46819d192290b0b/mono/metadata/icall-table.h#L482